tests/epyccel/modules/generic_functions_2.py

```diff
@@ -0,0 +1,171 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring/
+from pyccel.decorators import types
+from pyccel.decorators import template
+
+
+@template('z', types=['int', 'real'])
+@types('z', 'z')
+def tmplt_1(x, y):
+    return x + y

```

Comment:
Given that our unit tests also work as examples, would you mind capitalize the names of the template types throughout this file? E.g.
```python
@template('T', types=['int', 'real'])
@types('T', 'T')
```
This is a common convention for generic programming in C++